repair/row_level.cc

```diff
@@ -1152,6 +1161,11 @@ class repair_meta {
                 _master_node_shard_config.shard,
                 _seed,
                 std::invoke([this]() {
+                    if (_mixed_shard_optimization && !_is_tablet) {
+                        rlogger.debug("repair_reader: meta_id={}, read_strategy multishard_proxy is chosen for ks={}",
+                                _repair_meta_id, _schema->ks_name());
+                        return repair_reader::read_strategy::multishard_proxy;
```

Comment:
Could you please add a comment about how and why `multishard_proxy` is used, similarly to the comment on `multishard_filter` and `multishard_split` a few lines below?

Reply:
Yes, in multishard_proxy mode, a given token range is being worked on a single shard, it gets data from all the shards. The working shard is like a proxy for data from different shards.

Reply:
Thanks, but I was thinking about the comment in the code for the sake of people who will read it in the future. 
https://github.com/scylladb/scylladb/blob/ab8c3fcc789c37f0245b804715b313088f20b38a/repair/row_level.cc#L1177-L1192
The above significantly helped me with understanding the code. It would be nice to have something like this for the new strategy too.

Reply:
Yes, I will add comments. I am adding it here so that you could take a look before a new version is sent.